mechanics/physicell/include/physicell/agent_data.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <vector>
+
+#include <common/base_agent_data.h>
+#include <common/types.h>
+
+namespace physicore::mechanics::physicell {
+
+template <template <typename...> typename ContainerType = std::vector>
+struct mechanical_agent_data
+{
+public:
+	physicore::base_agent_data_generic_storage<ContainerType>& base_data;
+
+	index_t agents_count = 0;
+	index_t dims = 3;
+	index_t agent_types_count = 0;
+	index_t substrates_count = 0;
```

Comment:
I think that more reasonable default values for agent_types_count and substrates_count should be 1. otherwise we would be multiplying by zero and the array sizes would not make sense